lib/modules/platform/gitlab/index.ts

```diff
@@ -757,6 +757,44 @@ export async function createPr({
   }
   const description = sanitize(rawDescription);
   logger.debug(`Creating Merge Request: ${title}`);
+
+  let reviewerIds: number[] = [];
+  if (platformPrOptions?.gitLabReviewersFromApprovalRule) {
+    logger.debug(
+      `Fetching reviewers from GitLab approval rule: ${platformPrOptions.gitLabReviewersFromApprovalRule}`,
+    );
+    try {
+      const approvalRules = await gitlabApi.getJson(
+        `projects/${config.repository}/approval_rules`,
+        GitLabApprovalRules,
+      );
+
+      const approvalRuleReviewerIds = new Set<number>();
+      const matchingApprovalRule = approvalRules.body.find(
+        (rule) =>
+          rule.name === platformPrOptions.gitLabReviewersFromApprovalRule,
+      );
+
+      if (matchingApprovalRule?.eligible_approvers?.length) {
```

Comment:
Split this into 2 if's so that we can have separate log messages if:
- rule not found
- rule has no approvers